cmd/ci-operator-configresolver/main.go

```diff
@@ -0,0 +1,251 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/load"
+	"github.com/openshift/ci-tools/pkg/registry"
+	types "github.com/openshift/ci-tools/pkg/steps/types"
+)
+
+type options struct {
+	resolver   registry.Resolver
+	configPath string
+	regPath    string
+}
+
+type coalescer struct {
+	sync.Mutex
+	loading bool
+	loader  *sync.Cond
+	latest  *registry.Resolver
+}
+
+func (c *coalescer) reload(o options) {
```

Comment:
The options are invariant, let's choose which of them we need (the paths?) and make them part of the struct, not passed in every call